File: Sentinel-main/Sentinel-main/backend/squad_service.py
import requests
import os

SQUAD_SECRET_KEY = "sandbox_sk_0b63a4070d9916481e7c84b13fc250fdb4928933"
URL_BASE = "https://sandbox-api.squadco.com"

def initiate_escrow(amount, shipment_id):
    url = f"{URL_BASE}/transaction/initiate"
    headers = {"Authorization": f"Bearer {SQUAD_SECRET_KEY}"}
    payload = {
        "amount": int(amount * 100), #In kobo
        "email": "buyer@gmail.com",
        "currency": "NGN",
        "initiate_type": "inline",
        "transaction_ref": f"SENTINEL-{shipment_id}"
    }
    return requests.post(url, json=payload, headers=headers).json()

def payout_to_supplier(amount, bank, acc, shipment_id):
    url = f"{URL_BASE}/transfer"
    headers = {"Authorization": f"Bearer {SQUAD_SECRET_KEY}"}
    payload = {
        "amount": int(amount * 100), #In kobo
        "bank_name": bank,
        "account_number": acc,
        "transaction_ref": f"PAY-{shipment_id}",
        "remark": "Product Verified. Payment released."
    }
    return requests.post(url, json=payload, headers=headers).json()

File: Sentinel_2/src/services/squadService.js
const axios = require('axios');

const SQUAD_BASE_URL = 'https://sandbox-api-d.squadco.com'; // NOTE the '-d'
const SQUAD_SECRET_KEY = process.env.SQUAD_SECRET_KEY; 

const squadHeaders = {
    'Authorization': `Bearer ${SQUAD_SECRET_KEY}`,
    'Content-Type': 'application/json'
};


/**
 * Initiates an escrow payment
 * @param {number} amount - Amount in Naira
 * @param {string} email - Buyer's email
 * @param {string} shipmentId - Reference ID
 * @param {string} callbackUrl - Optional redirect URL after payment
 */
exports.initiatePayment = async (amount, email, shipmentId, callbackUrl = 'https://www.linkedin.com/') => {
    try {
        const payload = {
            amount: amount * 100, // Convert to kobo
            email: email,
            currency: "NGN",
            initiate_type: "inline",
            transaction_ref: `SENTINEL-${shipmentId}-${Date.now()}`,
            callback_url: callbackUrl
        };

        const response = await axios.post(`${SQUAD_BASE_URL}/transaction/initiate`, payload, { headers: squadHeaders });
        return response.data.data;
    } catch (error) {

        console.error('Squad Initiate Error:', error.response ? error.response.data : error.message);
        throw new Error(error.response?.data?.message || 'Failed to initiate payment');
    }
};


/**
 * Verifies a transaction
 * @param {string} transactionRef - Transaction reference
 */
exports.verifyPayment = async (transactionRef) => {
    try {
        const response = await axios.get(`${SQUAD_BASE_URL}/transaction/verify/${transactionRef}`, { headers: squadHeaders });
        return response.data.data;
    } catch (error) {
        console.error('Squad Verify Error:', error.response ? error.response.data : error.message);
        throw new Error(error.response?.data?.message || 'Failed to verify payment');
    }
};

/**
 * Transfers funds to a supplier (Payout)
 * @param {object} details - { amount, bank_code, account_number, account_name, transaction_reference }
 */
exports.releasePayout = async (details) => {
    try {
        const payload = {
            amount: details.amount * 100, // Convert to kobo
            bank_code: details.bank_code,
            account_number: details.account_number,
            account_name: details.account_name,
            transaction_reference: details.transaction_reference,
            currency_id: 'NGN',
            remark: 'Sentinel Escrow Payout'
        };

        const response = await axios.post(`${SQUAD_BASE_URL}/payout/transfer`, payload, { headers: squadHeaders });
        return response.data.data;
    } catch (error) {
        console.error('Squad Payout Error:', error.response ? error.response.data : error.message);
        throw new Error(error.response?.data?.message || 'Failed to release payout');
    }
};
